tools/upgrade.sh

```diff
@@ -2,6 +2,18 @@
 
 gateway_archive_url=$1
 node_modules_archive_url=$2
-wget $gateway_archive_url
-wget $node_modules_archive_url
-./tools/extract-release.sh ./gateway-*.tar.gz ./node_modules-*.tar.gz
+# wget $gateway_archive_url
+# wget $node_modules_archive_url
+
+pwd
+
+gateway_old=./gateway
+cp -r $gateway_old /tmp/gateway
+rm -r /tmp/gateway/node_modules
+$gateway_old/tools/extract-ca-archive.sh gateway-*.tar.gz /tmp
+$gateway_old/tools/extract-ca-archive.sh ../node_modules-*.tar.gz /tmp/gateway
+# TODO: will sudo ever prompt for password?
```

Comment:
The file: `/etc/sudodoers.d/010_pi-nopasswd` contains the following:
```
pi ALL=(ALL) NOPASSWD: ALL
```
which is what makes the pi user not need to enter a password.

Reply:
Great! Thanks for the info